seqr/views/apis/auth_api.py

```diff
@@ -31,10 +39,102 @@ def login_view(request):
     if not u:
         return create_json_response({}, status=401, reason='Invalid credentials')
 
+    request.session['anvil'] = None
+
     login(request, u)
 
     return create_json_response({'success': True})
 
+@csrf_exempt
+def login_google(request):
+  # Create flow instance to manage the OAuth 2.0 Authorization Grant Flow steps.
+  flow = google_auth_oauthlib.flow.Flow.from_client_secrets_file(
+      CLIENT_SECRETS_FILE, scopes=scopes)
+
+  # The URI created here must exactly match one of the authorized redirect URIs
+  # for the OAuth 2.0 client, which you configured in the API Console. If this
+  # value doesn't match an authorized URI, you will get a 'redirect_uri_mismatch'
+  # error.
+  flow.redirect_uri = 'http://localhost:3000/oauth2callback'
+
+  authorization_url, state = flow.authorization_url(
+      # Enable offline access so that you can refresh an access token without
+      # re-prompting the user for permission. Recommended for web server apps.
+      access_type='offline',
+      # Enable incremental authorization. Recommended as a best practice.
+      include_granted_scopes='true')
+
+  # Store the state so the callback can verify the auth server response.
+  request.session['state'] = state
+
+  # Return the url to be sent for the Google Auth server by the the front end
+  return create_json_response({'data': authorization_url})
+
+@csrf_exempt
+def login_oauth2callback(request):
+    # Specify the state when creating the flow in the callback so that it can
+    # verified in the authorization server response.
+    state = request.session['state']
+
+    flow = google_auth_oauthlib.flow.Flow.from_client_secrets_file(
+        CLIENT_SECRETS_FILE, scopes = scopes, state = state)
+    flow.redirect_uri = 'http://localhost:3000/oauth2callback'
+
+    # Use the authorization server's response to fetch the OAuth 2.0 tokens.
+    authorization_response = request.body.decode('utf-8')
+    flow.fetch_token(authorization_response = authorization_response)
+
+    token = flow.credentials.id_token
+
+    try:
+        # Decode the id token (It is a JWT token actually) to get user ID info
+        idinfo = id_token.verify_oauth2_token(token, requests.Request())
+    except ValueError as ve:
+        return create_json_response({}, status=401, reason=str(ve))
+
+    credentials = Credentials(**credentials_to_dict(flow.credentials))
+    session = AnvilSession(credentials = credentials, scopes = scopes)
+    try:
+        _ = session.get_anvil_profile()
```

Comment:
The function get_anvil_profile() will succeed only if the user has registered on AnVIL. It is for detecting user's registration on AnVIL.
Normally, the AnVIL profile includes user name, organization, and other user information. However, since we haven't been whitelisted, the information is just limited to the email address. After we are whitelisted, we should use the user names from the AnVIL profile rather than the names from the local User model.